gravitee-apim-rest-api/gravitee-apim-rest-api-management/gravitee-apim-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/auth/ExternalAuthenticationResource.java

```diff
@@ -0,0 +1,190 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.rest.api.management.rest.resource.auth;
+
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.proc.DefaultJWTClaimsVerifier;
+import io.gravitee.rest.api.idp.api.authentication.UserDetails;
+import io.gravitee.rest.api.management.rest.resource.auth.jwt.*;
+import io.gravitee.rest.api.management.rest.resource.auth.jwt.exceptions.InvalidAlgorithmException;
+import io.gravitee.rest.api.management.rest.resource.auth.jwt.exceptions.InvalidTokenException;
+import io.gravitee.rest.api.management.rest.resource.auth.jwt.jwks.ECJWKSourceResolver;
+import io.gravitee.rest.api.management.rest.resource.auth.jwt.jwks.MACJWKSourceResolver;
+import io.gravitee.rest.api.management.rest.resource.auth.jwt.jwks.RSAJWKSourceResolver;
+import io.gravitee.rest.api.management.rest.resource.auth.jwt.jwks.RemoteJWKSourceResolver;
+import io.gravitee.rest.api.management.rest.resource.auth.jwt.resolver.PublicKeyResolver;
+import io.gravitee.rest.api.model.UserEntity;
+import io.gravitee.rest.api.security.utils.AuthoritiesProvider;
+import io.gravitee.rest.api.service.common.GraviteeContext;
+import io.gravitee.rest.api.service.exceptions.UserNotFoundException;
+import jakarta.annotation.PostConstruct;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.QueryParam;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.Response;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import javax.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+@Singleton
+@Path("/auth/external")
+public class ExternalAuthenticationResource extends AbstractAuthenticationResource {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ExternalAuthenticationResource.class);
+    protected static final String ORG_CLAIM = "org";
+    protected static final String ENVIRONMENT_CLAIM = "env";
+    private boolean enabled;
+    private AbstractKeyProcessor<?> keyProcessor;
+    private Algorithm algorithm;
+
+    @Autowired
+    private AuthoritiesProvider authoritiesProvider;
+
+    private String uiURL;
+
+    @PostConstruct
+    public void afterPropertiesSet() {
+        enabled = environment.getProperty("auth.external.enabled", Boolean.class, false);
+
+        if (enabled) {
+            uiURL = environment.getProperty("installation.multi-tenant.accessPoints.organization.console", "http://localhost:3000");
```

Comment:
This property is a template containing place holder, also the xternal auth is not only associated to multi-tenant installation; You can find a specific service InstallationAccessQueryService, which should be used to retrieve the console URL of an organization. I suggest to rely on the existing cockpit auth resource as example.